### PR TITLE
Better AMOS error messages

### PR DESCRIPTION
**Pull request by musm**

### CODE REVIEW

File: src/bessel.jl [hunk 1]
```diff
@@ -3,8 +3,29 @@
...
+            "Input argument magnitude too large, complete loss of accuracy by argument reduction."
+        elseif id == 5
+            "Algorithm termination condition not met."
+        else
+            error("Invalid AMOS error flag")
+        end
+        new(id,msg)
+    end
 end
+Base.show(io::IO, err::AmosException) = print(io, "AmosException with id:$(err.id): $(err.msg)")
```
Comment (ararslan) [inline src/bessel.jl]: Should be `Base.showerror`. Also a test for this would be good, e.g. `@test sprint(showerror, AmosException(id)) == "something"`.

**Comment by musm** [comment]
I added the test you suggested, but is it really useful? Exceptions are already being tested, and that test would break if someone happened to change the error msg.

File: test/runtests.jl [hunk 2]
```diff
@@ -543,3 +543,5 @@ end
     @test typeof(SF.zeta(complex(1,1), 2f0)) == Complex{Float64}
     @test typeof(SF.zeta(complex(1), 2.0)) == Complex{Float64}
 end
+
+@test sprint(showerror, AmosException(1)) == "AmosException(1, \"Input error\")"
```
Comment (ararslan) [inline test/runtests.jl]: I don't think this is right based on how you've defined `showerror`

Comment (musm) [inline test/runtests.jl]: No it wasn't correct, I have updated.

**Comment by ararslan** [comment]
Yeah, not super useful, but good to have the `showerror` method tested. It doesn't get called for `@test_throws`.

Arguably, having that test break if someone changes the error message is a good thing, since it shows that the error message is what you'd expect it to be. That is, changing the error message on purpose should necessitate changing the test as well.

**Pull request merged**